src/engine/JoinHelpers.h

```diff
@@ -76,21 +85,29 @@ inline std::variant<LazyInputView, MaterializedInputView> resultToView(
   return convertGenerator(result.idTables(), permutation);
 }
 
-using GeneratorWithDetails =
-    cppcoro::generator<ad_utility::IdTableAndFirstCol<IdTable>,
-                       CompressedRelationReader::LazyScanMetadata>;
+// Type alias for the general InputRangeTypeErasedWithDetails with specific
+// types
+using GeneratorWithDetails = ad_utility::InputRangeTypeErasedWithDetails<
+    ad_utility::IdTableAndFirstCol<IdTable>,
+    CompressedRelationReader::LazyScanMetadata>;
 
 // Convert a `generator<IdTable` to a `generator<IdTableAndFirstCol>` for more
 // efficient access in the join columns below.
 inline GeneratorWithDetails convertGenerator(
     Permutation::IdTableGenerator gen) {
-  co_await cppcoro::getDetails = gen.details();
-  gen.setDetailsPointer(&co_await cppcoro::getDetails);
-  for (auto& table : gen) {
+  // Capture the details before moving the generator
+  auto details = gen.details();
+  auto transformer = [](auto& table) {
     // IndexScans don't have a local vocabulary, so we can just use an empty one
-    ad_utility::IdTableAndFirstCol t{std::move(table), LocalVocab{}};
-    co_yield t;
-  }
+    return ad_utility::IdTableAndFirstCol{std::move(table), LocalVocab{}};
+  };
+  auto cachingRange = ad_utility::CachingTransformInputRange(
+      std::move(gen), std::move(transformer));
+  auto typeErasedRange =
+      ad_utility::InputRangeTypeErased<ad_utility::IdTableAndFirstCol<IdTable>>(
+          std::move(cachingRange));
```

Comment:
This shouldn't be necessary, 
````
GeneratorWithDetails { CachingTransformRange(....), std::move(details)};
```
Should work directly (if not, then please tell me to assist with the constructors/deduction guides.